File: beecrowd3301.go
package main

import (
	"fmt"
	"sort"
)

type Duck struct {
	name string
	age  int
}

type duckList []Duck

func (d duckList) Len() int {
	return len(d)
}

func (d duckList) Less(i, j int) bool {
	return d[i].age < d[j].age
}

func (d duckList) Swap(i, j int) {
	d[i], d[j] = d[j], d[i]
}

func main() {
	ducks := []Duck{Duck{name: "huguinho"}, {name: "zezinho"}, {name: "luisinho"}}
	fmt.Scanf("%d %d %d", &ducks[0].age, &ducks[1].age, &ducks[2].age)
	sort.Sort(duckList(ducks))
	fmt.Println(ducks[1].name)
}

File: beecrowd3301.cpp
#include <bits/stdc++.h>
using namespace std;

struct Duck {
    string name;
    int age;

    bool operator < (const Duck& other) const {
        return age < other.age;
    }
};

int main() {
    Duck ducks[3] = {{"huguinho", 0}, {"zezinho", 0}, {"luisinho", 0}};
    cin >> ducks[0].age >> ducks[1].age >> ducks[2].age;
    sort(ducks, ducks + 3);
    cout << ducks[1].name << '\n';
}